app/scripts/background.js

```diff
@@ -101,6 +104,15 @@ const initApp = async (remotePort) => {
 
 if (isManifestV3) {
   browser.runtime.onConnect.addListener(initApp);
+  // Message below signals content script in DAPPS to connect to metamask background as backend is not active
+  // It is required to re-connect DAPPS after service worker re-activation
+  chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
```

Comment:
Note that we should be checking for errors after any extension API comment (e.g. check `browser.runtime.lastError`), because that's how errors are "throw" for the Chrome extension API. That applies here and to `sendMessage`

Reply:
@Gudahtt 

added checks for `lastError` here https://github.com/MetaMask/metamask-extension/pull/16075/commits/bbd5cffde0503e41e1056a045d2912981675af18